FrEee/Scripts/AI/Vanguard/MinistryOfInfrastructure.csx

```diff
@@ -0,0 +1,246 @@
+    protected FacilityTemplate GetPlanetVitalFacilities(Planet planet)
+    {
+        //TODO: handle things like planet increasing planets. 
+        return null; 
+    }
+
+    /// <summary>
+    /// Chooses what resource facility to build on this planet. 
+    /// </summary>
+    /// <param name="planet"></param>
+    /// <returns></returns>
+    protected FacilityTemplate GetResourceFacility(Planet planet, int randMin = 10, int randMax = 100)
+    {
+        return (from faci in AvailableFacilities
+        where (faci.HasAbility("Resource Generation - Minerals")
+           || faci.HasAbility("Resource Generation - Organics")
+           || faci.HasAbility("Resource Generation - Radioactives")
+           || faci.HasAbility("Point Generation - Research")
+           || faci.HasAbility("Point Generation - Intelligence"))
+           //ensure a little random in the scores, so not always building what is best. 
+        orderby GetScoreForFacility(planet, faci) + RandomHelper.Range(randMin, randMax) descending
+        select faci).FirstOrDefault(); 
+
+            
+    }
+
+    /// <summary>
+    /// Gets the score for the given facility. 
+    /// </summary>
+    /// <param name="planet"></param>
+    /// <param name="facilityTemplate"></param>
+    /// <returns></returns>
+    int GetScoreForFacility(Planet planet, FacilityTemplate facilityTemplate)
+    {
+        int score = 0;
+
+        var prefix = "Resource Generation - ";
+        var pcts = planet.StandardIncomePercentages;
+        foreach (var abil in facilityTemplate.Abilities().Where(abil => abil.Rule.Name.StartsWith(prefix)))
+        {
+            var resourceName = abil.Rule.Name.Substring(prefix.Length);
+            var resource = Resource.Find(resourceName);
+            var amount = abil.Value1.ToInt();
+
+            if (resource.HasValue)
+                amount = Galaxy.Current.StandardMiningModel.GetRate(amount, planet.ResourceValue[resource], pcts[resource] / 100d);
+
+            score += amount + GetResourceScoreMultiplier(resource); 
+        }
+        prefix = "Point Generation - ";
+        foreach (var abil in facilityTemplate.Abilities().Where(abil => abil.Rule.Name.StartsWith(prefix)))
+        {
+            var resoruceName = abil.Rule.Name.Substring(prefix.Length);
+            var resource = Resource.Find(resoruceName);
+            var amount = abil.Value1.ToInt() * pcts[resource] / 100;
+
+            score += amount + GetResourceScoreMultiplier(resource); 
+        }
+
+        return score; 
+    }
+
+    /// <summary>
+    /// Modifies the default score of a facility by the current aspects of the empire. 
+    /// </summary>
+    /// <param name="resourceName"></param>
+    /// <returns></returns>
+    int GetResourceScoreMultiplier(Resource resourceName)
+    {
+        var income = Empire.NetIncomeLessConstruction[resourceName]; 
+        switch(resourceName.Name)
+        {
+            case "Minerals":
+                if (income < 0)
```

Comment:
Any reason you're using hardcoded thresholds rather than some sort of sliding function?

Reply:
Was easier to code honestly. That can be tweaked later if needed.